core/ir/instr_api.h

```diff
@@ -1898,13 +1898,17 @@ instr_is_rep_string_op(instr_t *instr);
  */
 typedef enum {
     DR_INSTR_CATEGORY_UNCATEGORIZED = 0x0, /**< Uncategorized. */
-    DR_INSTR_CATEGORY_INT_MATH = 0x1,      /**< Integer arithmetic operations. */
-    DR_INSTR_CATEGORY_FP_MATH = 0x2,       /**< Floating-Point arithmetic operations. */
+    DR_INSTR_CATEGORY_INT = 0x1,           /**< Integer arithmetic operations. */
+    DR_INSTR_CATEGORY_FP = 0x2,            /**< Floating-Point operations. */
     DR_INSTR_CATEGORY_LOAD = 0x4,          /**< Loads. */
     DR_INSTR_CATEGORY_STORE = 0x8,         /**< Stores. */
     DR_INSTR_CATEGORY_BRANCH = 0x10,       /**< Branches. */
-    DR_INSTR_CATEGORY_SIMD = 0x20, /**< Operations with vector registers (SIMD). */
-    DR_INSTR_CATEGORY_OTHER = 0x40 /**< Other types of instructions. */
+    DR_INSTR_CATEGORY_SIMD = 0x20,     /**< Operations with vector registers (SIMD). */
+    DR_INSTR_CATEGORY_STATE = 0x40,    /**< Loads, stores, or queries processor state. */
```

Comment:
To avoid confusion, how about "Saves, restores, or queries processor state."  Then nobody will think it's a regular load or store.  (Make the same change to the FP category description too, please.)